srcpkgs/libco/template

```diff
@@ -0,0 +1,36 @@
+# Template file for 'libco'
+pkgname=libco
+version=1.0.1
+revision=1
+short_desc="Cooperative multithreading library written in C89"
+maintainer="Julio Galvan <julio@epazote.net>"
+license="ISC"
+homepage="https://github.com/freeekanayaka/libco"
+distfiles="https://github.com/freeekanayaka/libco/archive/v${version}.tar.gz"
+checksum=b5a49b0352e52edffc0046b592e4414945a2b637f7433f8c526b8cdaab9d24bd
+shlib_provides="libco.so"
```

Comment:
pretty sure that you can remove this

Reply:
without this line, building lxd gives trouble
```
...
=> ERROR: lxd-3.15_1: failed to install 'lxc-devel-3.1.0_1' dependency! (error 8)
dqlite-0.9.1_1: broken, unresolvable shlib `libco.so'
Transaction aborted due to unresolved shlibs.
=> ERROR: Please see above for the real error, exiting...
```

Reply:
You should not need this line given that you have added libco to the common/shlibs file.  Please remove and re-test.

Reply:
I have just tested with the line removed. libco package is built without the shlibs provides property set correctly, it does not provide libco.so without the line.

Reply:
The thing is, it should provide the versioned one, not `libco.so` per se...

Reply:
I did not notice a versioned one in the github repo. Does upstream libco provide a versioned library?

Reply:
It does not. But it should - if it does not there is no point in versioning it manually in the template (the linkage will always be wrong). I reported this in the upstream repo

Reply:
This is why:

```
q66@moca: /home/q66$ gcc -fPIC foo.c -shared -o libfoo.so.1
q66@moca: /home/q66$ ln -s /home/q66/libfoo.so.1 libfoo.so
q66@moca: /home/q66$ gcc test.c -o test -L. -lfoo
q66@moca: /home/q66$ ldd test
        linux-vdso64.so.1 (0x00003fa1975d0000)
        libfoo.so => not found
        libc.so.6 => /usr/lib/libc.so.6 (0x00003fa1973a9000)
        /lib/ld64.so.2 (0x00003fa1975d8000)
q66@moca: /home/q66$ gcc -fPIC foo.c -shared -o libfoo.so.1 -Wl,-soname,libfoo.so.1
q66@moca: /home/q66$ gcc test.c -o test -L. -lfoo
q66@moca: /home/q66$ ldd test
        linux-vdso64.so.1 (0x00003f7af9820000)
        libfoo.so.1 => not found
        libc.so.6 => /usr/lib/libc.so.6 (0x00003f7af95f9000)
        /lib/ld64.so.2 (0x00003f7af9833000)
```

Reply:
If you version it manually just by creating symlinks this also means that once you don't have the devel package installed the dynamic linker will fail to find the runtime library afterwards. So at very least you would also need to provide the appropriate linker flag and then remove the `shlib_provides`.